http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,219 @@
+package io.quarkus.ts.http.advanced.reactive;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.testcontainers.shaded.org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.http.HttpStatus;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.netty.handler.codec.http2.Http2Exception;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.quarkus.test.services.URILike;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+@QuarkusScenario
+@ExtendWith(VertxExtension.class)
+public class Http2IT {
+    private static final Logger LOG = Logger.getLogger(Http2IT.class);
+    @QuarkusApplication(classes = { MorningResource.class }, properties = "http2.properties")
+    static RestService app = new RestService();
+
+    private static URILike baseUri;
+    static private Vertx vertx;
+
+    private static final String BASE_ENDPOINT = "/morning";
+    private static final String GREETING = "Buenos dias";
+
+    private static HttpClient httpClient;
+
+    @BeforeAll
+    static void setUp() {
+        baseUri = app.getURI();
+        vertx = Vertx.vertx();
+    }
+
+    @Test
+    @DisplayName("HttpClient Vertx compatibility HTTP/1.1 Test")
+    public void httpClientVertxHttp1(VertxTestContext vertxTestContext) {
+        Checkpoint requestCheckpoint = vertxTestContext.checkpoint(2);
+        assertNotNull(vertx);
+        httpClient = vertx.createHttpClient();
+        assertNotNull(httpClient);
+        String expectedHTTP1Version = "HTTP_1_1";
+
+        vertxTestContext.verify(() -> httpClient.request(HttpMethod.GET, baseUri.getPort(), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(req -> req.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(expectedHTTP1Version, httpClientResponse.version().toString());
+                            requestCheckpoint.flag();
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> {
+                    LOG.info("Got data " + body.toString("ISO-8859-1"));
```

Comment:
Personally I'd only be interested in this information if something goes wrong, like if the assertion fails you should log it as part of the reason of assertion failure. Apart of logging being expensive (which is no big deal here), I just think how would tests look like if everyone logged information like this. We are in a different situation from production because we can reproduce things and if we can't, we have the information logged on failure.

Reply:
Same goes to other logging in this class.

Reply:
Yes, I agree, sometimes I do some "`System.out.println`", in the tests to check that everything is working as expected, I changed to LOG for best practices, but that's all.